src/python/pants/backend/python/lint/yapf/rules_integration_test.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import pytest
+
+from pants.backend.python.lint.yapf.rules import YapfFieldSet, YapfRequest
+from pants.backend.python.lint.yapf.rules import rules as yapf_rules
+from pants.backend.python.target_types import PythonLibrary
+from pants.core.goals.fmt import FmtResult
+from pants.core.goals.lint import LintResult, LintResults
+from pants.core.util_rules import config_files, source_files
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.addresses import Address
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.target import Target
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *yapf_rules(),
+            *source_files.rules(),
+            *config_files.rules(),
+            QueryRule(LintResults, (YapfRequest,)),
+            QueryRule(FmtResult, (YapfRequest,)),
+            QueryRule(SourceFiles, (SourceFilesRequest,)),
+        ],
+        target_types=[PythonLibrary],
+    )
+
+
+GOOD_FILE = "data = []\n"
+BAD_FILE = "data = {  'a':11,'b':22    }\n"
+FIXED_BAD_FILE = "data = {'a': 11, 'b': 22}\n"
+
+# Note the indentation is 6 spaces; after formatting it should become 2
+NEEDS_FORMATTING_FILE = "def func():\n      return 42\n"
+FIXED_NEEDS_FORMATTING_FILE_INDENT2 = "def func():\n  return 42\n"
```

Comment:
Do you mean CONFIG, rather than FORMATTING? The idea here is that behavior changes based on config, which we use to validate that we did properly wire things up to the underlying tool.

Reply:
Hm, no, I meant `FORMATTING` :) I meant to describe that here is the file that needs to be formatted (it hasn't been formatted yet). Do you think this is not clear / confusing to the reader? I am happy to rename if you think it is.

Reply:
I think it's not clear how this differs from BAD_FILE. 

Also, we want a file that specifically needs config for certain behavior to be triggered. Meaning, either:
- it normally passes, but with config will fail.
- it normally fails, but with config will pass.

This is how we can validate that config files and passthrough args are actually being used.

Reply:
Right, sorry about that, I think I got it now. I've renamed the variable which should hopefully make it clearer that this file needs a config to be formatted and depending on the passed config (either via a file or via passthrough arguments), will be formatted in a certain way. This should prove that config files and passthrough args are actually being used.